python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -536,45 +543,57 @@ def _internal_job_finalisation(job, updated_dirac_status):
         logger = getLogger()
 
         if updated_dirac_status == 'completed':
+            import time
+            start = time.time()
             # firstly update job to completing
             DiracBase._getStateTime(job, 'completing')
             if job.status in ['removed', 'killed']:
                 return
-            if (job.master and job.master.status in ['removed', 'killed']):
+            elif (job.master and job.master.status in ['removed', 'killed']):
```

Comment:
No strong reason other than the fact I had an else function with some logging but this got dropped and I just stuck with what was left.